src/event/AsyncEventTrait.php

```diff
@@ -24,61 +24,17 @@
 namespace pocketmine\event;
 
 use pocketmine\promise\Promise;
-use pocketmine\promise\PromiseResolver;
-use pocketmine\utils\ObjectSet;
-use function array_shift;
-use function count;
 
 trait AsyncEventTrait {
-	/** @phpstan-var ObjectSet<Promise<null>> */
-	private ObjectSet $promises;
+	private AsyncEventDelegate $delegate;
```

Comment:
Beware GC performance impact caused by circular references. Is it even necessary to store this field in the event object?

Reply:
That can be solved if `AsyncEventDelegate` stores a `WeakReference` of the class.

Reply:
The point is that the circular reference was never necessary to begin with. Why store an object you don't use?

Reply:
Right. That class can just be `AsyncEventTrait`.

Reply:
Right. That class can just be `AsyncEventTrait`.

Reply:
> The point is that the circular reference was never necessary to begin with. Why store an object you don't use?

I agree, the class property is not necessary